src/vuln_analysis/utils/functions_parsers/lang_functions_parsers.py

```diff
@@ -106,3 +108,23 @@ def get_function_reserved_word(self) -> str:
     @abstractmethod
     def get_type_reserved_word(self) -> str:
         pass
+
+    @abstractmethod
+    def get_dummy_function(self, function_name):
+        pass
+
+    @staticmethod
+    def is_script_language():
+        return False
+
+    @staticmethod
+    @abstractmethod
+    def is_same_package(package_name_from_input, package_name_from_tree):
+        pass
+
+    @staticmethod
+    def get_builder_function_name():
+        return None
```

Comment:
@TamarW0 This is a great Idea.
but the name is a bit confusing...
```suggestion
    @staticmethod
    def get_constructor_method_name():
        return None
```
Please refactor across all places.
This way, It will be more readable, and more targeted for its purpose
( in python the constructor method is always `__init__` , in javascript it's always `constructor` , and in java it's the class name , etc).